### PR TITLE
LPD-43420 migrate some Categories.testcase to integration tests

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-43420 Migrate AddCategoryWithOwnerViewPermission, AddCategoryWithSiteMemberViewPermission, AddCategoryWithGuestViewPermission to Integration test

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-43420

To make the CI faster, is better to use integration/unit test than Poshi test.

# How am I fixing it?

Creating a test that summarize the behaviour that the POSHI tests were testing

# How can you verify that it works?

Run the included automated tests.

# Commits

- **LPD-43420 migrate AddCategoryWithGuestViewPermission, AddCategoryWithOwnerViewPermission, and AddCategoryWithSiteMemberViewPermission to integration tests**
- **LPD-43420 refactor test extract common code**

### CODE REVIEW

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java [hunk 1]
```diff
@@ -70,6 +72,66 @@ public void setUp() throws Exception {
 			ServiceContextTestUtil.getServiceContext());
 	}
 
+	@Test
+	public void testAddCategoryWithGuestViewPermission() throws Exception {
```
Comment (jkappler) [inline modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java]: @AliciaGarciaGarcia, why not create a private method called _testAddCategoryWithViewPermission and pass the RoleConstant?

**Review by jkappler — CHANGES_REQUESTED** [review]
See my comment, we'll talk tomorrow ❤️

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java]: no strong opinion.

I'll rebase this and resend :D

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java [hunk 2]
```diff
@@ -128,6 +146,20 @@ public void testSearchWithoutParentGroupPermission() throws Exception {
 		}
 	}
 
+	private void _testAddCategoryWithViewPermission(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryServiceTest.java:149]: I will make same changes asked than on https://github.com/liferay-content-management/liferay-portal/pull/6156#discussion_r1871540655

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
my bad.

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**